datajoint/table.py

```diff
@@ -366,6 +370,10 @@ def insert(
                 drop=len(rows.index.names) == 1 and not rows.index.names[0]
             ).to_records(index=False)
 
+        if isinstance(rows, Path):
+            with open(rows, newline="") as data_file:
+                rows = list(csv.DictReader(data_file, delimiter=","))
```

Comment:
Awesome.  

Is `list` required here? Would it not work with just the `csv.DictReader` alone?

Reply:
Leaving it as a `csv.DictReader` object here results in errors later when we go to read the rows outside of the `open` context
```python
File ~/Documents/dev/datajoint-python/datajoint/table.py:418, in <genexpr>(.0)
    415     return
    417 field_list = []  # collects the field list from first row (passed by reference)
--> 418 rows = list(
    419     self.__make_row_to_insert(row, field_list, ignore_extra_fields)
    420     for row in rows
    421 )
    422 if rows:
    423     try:

ValueError: I/O operation on closed file.
```

Reply:
Oh, that's because the `insert` is outside the `with` clause and the file has been closed. 

It's okay to leave as is and read into a list before inserting.